### PR TITLE
CDRIVER-3786 clean up OCSP docs

### PR DESCRIPTION
**Pull request by kevinAlbs**
- only note OCSP behavior exceptions
- note that crl_file is unsupported with LibreSSL and Secure Transport

### CODE REVIEW

File: src/libmongoc/doc/configuring_tls.rst [hunk 1]
```diff
@@ -88,25 +88,21 @@ Ensure your system's OpenSSL is a recent version (at least 1.0.1), or install a
...
 `````````````````
 
 The MongoDB C Driver supports LibreSSL through the use of OpenSSL compatibility checks when configured to compile against ``openssl``. It also supports the new ``libtls`` library when configured to build against ``libressl``.
 
+When compiled against the Windows native libraries, the ``crl_file`` option of a :symbol:`mongoc_ssl_opt_t` is not supported, and will issue an error if used.
+
 Setting ``tlsDisableOCSPEndpointCheck`` and ``tlsDisableCertificateRevocationCheck`` has no effect.
 
-The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_).
+The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_) with the following notes:
```
Comment (bazile-clyde) [inline src/libmongoc/doc/configuring_tls.rst:104]:
```suggestion
The Online Certificate Status Protocol (OCSP) (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_)  is partially supported with the following notes:
```
nit: Maybe switch these phrases and below for consistency with the OpenSSL change above? Completely optional though.

File: src/libmongoc/doc/configuring_tls.rst [hunk 2]
```diff
@@ -23,7 +23,7 @@ The following URI options may be used to further configure TLS:
 Configuration with mongoc_ssl_opt_t
 -----------------------------------
 
-Alternatively, the :symbol:`mongoc_ssl_opt_t` struct may be used to configure TLS with :symbol:`mongoc_client_set_ssl_opts()` or :symbol:`mongoc_client_pool_set_ssl_opts()`. Most of the configurable options can be using the Connection URI.
+Alternatively, the :symbol:`mongoc_ssl_opt_t` struct may be used to configure TLS with :symbol:`mongoc_client_set_ssl_opts()` or :symbol:`mongoc_client_pool_set_ssl_opts()`. Most of the configurable options can be set using the `Connection String URI <http://docs.mongodb.org/manual/reference/connection-string/>`_.
```
Comment (jmikola) [inline src/libmongoc/doc/configuring_tls.rst:26]: Nit, but we should probably use `https` whenever possible.

File: src/libmongoc/doc/configuring_tls.rst [hunk 3]
```diff
@@ -57,9 +57,9 @@ Server Certificate Verification
 
 The MongoDB C Driver will automatically verify the validity of the server certificate, such as issued by configured Certificate Authority, hostname validation, and expiration.
 
-To overwrite this behaviour, it is possible to disable hostname validation, OCSP endpoint revocation checking, revocation checking entirely, and allow invalid certificates.
+To overwrite this behavior, it is possible to disable hostname validation, OCSP endpoint revocation checking, revocation checking entirely, and allow invalid certificates.
```
Comment (jmikola) [inline src/libmongoc/doc/configuring_tls.rst:60]: Don't tell @derickr.

File: src/libmongoc/doc/configuring_tls.rst [hunk 4]
```diff
@@ -123,13 +119,10 @@ When ``tlsCAFile`` is set, the driver will only allow server certificates issued
 
 When ``crl_file`` is set with :symbol:`mongoc_ssl_opt_t`, the driver will import the revocation list to the ``System Local Machine Root`` certificate store.
 
-Setting ``tlsDisableCertificateRevocationCheck`` disables certificate revocation checking.
-
 Setting ``tlsDisableOCSPEndpointCheck`` has no effect.
 
-The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_).
+The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_) with the following notes:
```
Comment (jmikola) [inline src/libmongoc/doc/configuring_tls.rst:124]: Same as @bazile-clyde's comment above.

File: src/libmongoc/doc/configuring_tls.rst [hunk 5]
```diff
@@ -141,14 +134,13 @@ Native TLS Support on macOS / Darwin (Secure Transport)
...
 
-When compiled against Secure Transport, the ``ca_dir`` option of a :symbol:`mongoc_ssl_opt_t` is not supported, and will issue an error if used.
+When compiled against Secure Transport, the ``ca_dir`` and ``crl_file`` options of a :symbol:`mongoc_ssl_opt_t` are not supported. An error is issued if either are used.
 
 When ``tlsCAFile`` is set, the driver will only allow server certificates issued by the authority (or authorities) provided. When no ``tlsCAFile`` is set, the driver will use the Certificate Authorities in the currently unlocked keychains.
 
 Setting ``tlsDisableOCSPEndpointCheck`` and ``tlsDisableCertificateRevocationCheck`` has no effect.
 
-The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_).
+The Online Certificate Status Protocol (OCSP) is partially supported (see `RFC 6960 <https://tools.ietf.org/html/rfc6960>`_) with the following notes.
```
Comment (jmikola) [inline src/libmongoc/doc/configuring_tls.rst:143]: Same as @bazile-clyde's comment above.

File: src/libmongoc/doc/configuring_tls.rst (hunk 1 above)
Comment (kevinAlbs) [reply src/libmongoc/doc/configuring_tls.rst:104]: Done.

File: src/libmongoc/doc/configuring_tls.rst (hunk 2 above)
Comment (kevinAlbs) [reply src/libmongoc/doc/configuring_tls.rst]: Good point. I'll do a global find-replace for `docs.mongodb.org` links in a follow-up commit. Though at least docs.mongodb.org has redirects in place at least.

File: src/libmongoc/doc/configuring_tls.rst (hunk 4 above)
Comment (kevinAlbs) [reply src/libmongoc/doc/configuring_tls.rst]: Done.

File: src/libmongoc/doc/configuring_tls.rst (hunk 5 above)
Comment (kevinAlbs) [reply src/libmongoc/doc/configuring_tls.rst]: Done.

**Pull request merged**